x-pack/plugins/apm/server/lib/rum_client/get_page_load_distribution.ts

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { getRumOverviewProjection } from '../../../common/projections/rum_overview';
+import { mergeProjection } from '../../../common/projections/util/merge_projection';
+import {
+  Setup,
+  SetupTimeRange,
+  SetupUIFilters,
+} from '../helpers/setup_request';
+
+export async function getPageLoadDistribution({
+  setup,
+  minPercentile,
+  maxPercentile,
+}: {
+  setup: Setup & SetupTimeRange & SetupUIFilters;
+  minPercentile?: string;
+  maxPercentile?: string;
+}) {
+  const projection = getRumOverviewProjection({
+    setup,
+  });
+
+  const params = mergeProjection(projection, {
+    body: {
+      size: 0,
+      query: {
+        bool: projection.body.query.bool,
+      },
+      aggs: {
+        durationMinMax: {
+          min: {
+            field: 'transaction.duration.us',
+            missing: 0,
+          },
+        },
+        durationPercentiles: {
+          percentiles: {
+            field: 'transaction.duration.us',
+            percents: [50, 75, 90, 95, 99],
+            script: {
+              lang: 'painless',
+              source: "doc['transaction.duration.us'].value / params.timeUnit",
+              params: {
+                timeUnit: 1000,
+              },
+            },
```

Comment:
out of curiosity, why a painless script instead of converting on the Kibana server?

I'd also recommend setting `hdr.number_of_significant_value_digits` as it's faster. See https://github.com/elastic/kibana/issues/62399

Reply:
My naive assumption was that anything that's done on java is probably faster than JS server :D 
but i think it probably hardly makes a difference? i think in this case probably a cleaner code makes more sense. 
though i like the option `hdr.number_of_significant_value_digits` i must have overlooked that.

Reply:
The script is applied before the percentiles are calculated, so on each value. So the number of times you have to execute this piece of code will be (way) higher in ES. In Kibana you would only apply this transformation on the returned percentiles.

Btw, `number_of_significant_value_digits` applies to accuracy, it won't apply any formatting for you (IIRC).